File: app/src/main/java/com/simbirsoft_mobile_revolution2017_v_team/mobilelibrary/domain/BookBuilder.java
package com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.domain;

/**
 * Created by Olegka on 12.10.2017.
 */

public class BookBuilder {

    private String id;
    private String name;
    private String author;
    private long year;
    private String publishingHouse;
    private String ISBN;
    private int numberOfPages;
    private boolean isAvailable;
    private boolean wasReaded;
    private boolean isFavourite;

    public BookBuilder id(String id) {
        this.id = id;
        return this;
    }

    public BookBuilder name(String name) {
        this.name = name;
        return this;
    }

    public BookBuilder author(String author) {
        this.author = author;
        return this;
    }

    public BookBuilder year(long year) {
        this.year = year;
        return this;
    }

    public BookBuilder publishingHouse(String publishingHouse) {
        this.publishingHouse = publishingHouse;
        return this;
    }

    public BookBuilder ISBN(String ISBN) {
        this.ISBN = ISBN;
        return this;
    }

    public BookBuilder numberOfPages(int numberOfPages) {
        this.numberOfPages = numberOfPages;
        return this;
    }

    public BookBuilder isAvailable(boolean isAvailable) {
        this.isAvailable = isAvailable;
        return this;
    }

    public BookBuilder wasReaded(boolean wasReaded) {
        this.wasReaded = wasReaded;
        return this;
    }

    public BookBuilder isFavourite(boolean isFavourite) {
        this.isFavourite = isFavourite;
        return this;
    }

    public Book build() {
        return new Book(this);
    }

    public String getId(){
        return id;
    }

    public String getName() {
        return name;
    }

    public String getAuthor() {
        return author;
    }

    public long getYear() {
        return year;
    }

    public String getPublishingHouse() {
        return publishingHouse;
    }

    public String getISBN() {
        return ISBN;
    }

    public int getNumberOfPages() {
        return numberOfPages;
    }

    public boolean isAvailable() {
        return isAvailable;
    }

    public boolean isWasReaded() {
        return wasReaded;
    }

    public boolean isFavourite() {
        return isFavourite;
    }

}

File: app/src/main/java/com/simbirsoft_mobile_revolution2017_v_team/mobilelibrary/domain/BookBuilder.kt
package com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.domain

/**
 * Created by user on 14.11.2017.
 */

class BookBuilderK {
    var id: String? = null
        private set
    var name: String? = null
        private set
    var author: String? = null
        private set
    var year: Long = 0
        private set
    var publishingHouse: String? = null
        private set
    var isbn: String? = null
        private set
    var numberOfPages: Int = 0
        private set
    var isAvailable: Boolean = false
        private set
    var wasReaded: Boolean = false
        private set
    var isFavourite: Boolean = false
        private set

    fun id(id : String): BookBuilderK {
        this.id = id
        return this
    }

    fun name(name: String): BookBuilderK {
        this.name = name
        return this
    }

    fun author(author: String): BookBuilderK {
        this.author = author
        return this
    }

    fun year(year: Long): BookBuilderK {
        this.year = year
        return this
    }

    fun publishingHouse(publishingHouse: String): BookBuilderK {
        this.publishingHouse = publishingHouse
        return this
    }

    fun ISBN(ISBN: String): BookBuilderK {
        this.isbn = ISBN
        return this
    }

    fun numberOfPages(numberOfPages: Int): BookBuilderK {
        this.numberOfPages = numberOfPages
        return this
    }

    fun isAvailable(isAvailable: Boolean): BookBuilderK {
        this.isAvailable = isAvailable
        return this
    }

    fun wasReaded(wasReaded: Boolean): BookBuilderK {
        this.wasReaded = wasReaded
        return this
    }

    fun isFavourite(isFavourite: Boolean): BookBuilderK {
        this.isFavourite = isFavourite
        return this
    }

    fun build(): BookK = BookK(this)
}